lib/manager/components/FeedSourceTableRow.js

```diff
@@ -241,31 +266,56 @@ class FeedInfo extends PureComponent<{ feedSource: Feed }> {
         </h4>
         <h5>{lastUpdateText}</h5>
         <Row>
-          {feedSource.url &&
+          {feedSource.url && (
             <Col xs={12}>
               <Icon type='link' />
               <a href={feedSource.url} target='_blank'>
                 {abbreviate(feedSource.url)}
                 <Icon type='external-link' />
               </a>
             </Col>
-          }
-          {feedSource.retrievalMethod === 'FETCHED_AUTOMATICALLY' &&
+          )}
+          {feedSource.retrievalMethod === 'FETCHED_AUTOMATICALLY' && (
             <Col xs={6}>
               <Icon type='feed' />
               Auto fetch
             </Col>
-          }
-          {feedSource.deployable &&
+          )}
+          {feedSource.deployable && (
             <Col xs={6}>
               <Icon type='rocket' />
               Deployable
             </Col>
-          }
+          )}
           <Col xs={6}>
             <Icon type={feedSource.isPublic ? 'globe' : 'lock'} />
             {feedSource.isPublic ? 'Public' : 'Private'}
           </Col>
+          <Col xs={12}>
+            <div className='feedSourceLabelRow'>
+              {project.labels.length > 0 && (
+                <Icon type={'tag'} />
+              )}
+              <LabelAssignerModal
+                ref='labelAssignerModal'
+                feedSource={feedSource}
+                project={project}
+              />
+              {userCanManageFeed && project.labels.length > 0 && (
+                <button
+                  className='labelActionButton small'
+                  style={{ marginRight: '5px' }}
+                  onClick={() => this.refs.labelAssignerModal.open()}
+                >
+                  Edit
+                </button>)}
+              <div className='feedLabelContainer'>
+                {feedSource.labels.map((label) => (
+                  <FeedLabel small key={label.id} label={label} />
+                ))}
```

Comment:
We may have to discuss the layout and label text size in here. Do we prefer labels so they flow like text? (as opposed to vertically stacked?)

Reply:
I personally prefer the flowing labels, mostly because they save space. However, with the new `grid` class I added, it's easy to switch! Depends on what we want @binh-dam-ibigroup @landonreed